flux/src/main/java/software/amazon/aws/clients/swf/flux/poller/PartitionMetadata.java

```diff
@@ -64,12 +75,103 @@ public static PartitionMetadata fromPartitionIdGeneratorResult(PartitionIdGenera
+        // Most of the time, the entire thing will fit in one marker, so we'll generate that and return it if it's short enough.
+        // We do this because calculating the actual size one partition ID at a time is relatively expensive.
+
+        // We'll leave ourselves a little wiggle room for our attribute names, and to simplify the logic below.
+        // This way we can overrun our length limit by one partition id without it being a problem.
+        // The actual max enforced by SWF is 32768, and partition IDs can't be longer than 250 characters anyway, since
+        // the activity ID would get too long.
+        int maxLength = 32000;
+
+        String fullJson = MAPPER.writeValueAsString(this);
+        if (fullJson.length() <= maxLength) {
+            return Collections.singletonList(fullJson);
+        }
+
+        // the base json looks like this, which is 52 characters long:
+        // {"partitionIds":[],"encodedAdditionalAttributes":{}}
+        int baseJsonSize = 52;
+
+        Map<String, String> attributes = encodedAdditionalAttributes;
+        int attributeMapSize = MAPPER.writeValueAsString(attributes).length();
+
+        List<String> subsets = new LinkedList<>();
+
+        // We'll use a TreeSet here even though it doesn't really need to be sorted in the output just so testing is easier.
+        Set<String> partitionIdSubset = new TreeSet<>();
+        int partitionIdSubsetSize = 0;
+        for (String partitionId : partitionIds) {
+
+            // This serialization includes the quotes around the partition IDs, and we need to account for a comma after each.
+            // We can't just do a naive "length plus three"; there might be escaped characters in the serialized identifier.
+            // This technically means we're counting one extra comma than we will actually use but this shouldn't cause
+            // any extra markers to be generated since it only matters for the last partition ID, and we allow exceeding maxLength
+            // for the last partition ID in each subset.
+            String serialized = MAPPER.writeValueAsString(partitionId);
+            partitionIdSubsetSize += serialized.length() + 1;
+
+            // We add the partition ID here, not the serialized version, since we'll properly re-serialize the partition metadata
+            // as a whole once we have enough partition IDs.
+            partitionIdSubset.add(partitionId);
+
+            if (partitionIdSubsetSize + attributeMapSize + baseJsonSize >= maxLength) {
+                PartitionMetadata subset = new PartitionMetadata(partitionIdSubset, attributes);
+                subsets.add(MAPPER.writeValueAsString(subset));
+
+                partitionIdSubset.clear();
+                partitionIdSubsetSize = 0;
+
+                // This ensures only the first metadata marker contains the attributes.
+                attributes = Collections.emptyMap();
+                attributeMapSize = 0;
```

Comment:
nit. This isn't true, it would be 2 chars since an empty map gets serialized as `{}`

Reply:
That's included in `baseJsonSize`.

Reply:
Now that you've pointed it out though, I realize I'm technically counting those characters twice for the first page :)  I'll fix it.